### PR TITLE
--genome for hicAggregateContacts by samuelCollombet

### PR DESCRIPTION
**Pull request by gtrichard**
* [ ] Flake8 passes (`flake8 . --exclude=.venv,.build,planemo_test_env,build --ignore=E501,F403,E402,F999,F405,E712`)
* [ ] Local tests pass (`py.test hicexplorer --doctest-modules`)

### CODE REVIEW

**Comment by joachimwolff** [comment]
Thanks for your contribution.

The basis you took to apply your changes was outdated, please merge first the current release22beta branch into your branch, apply than the changes and make sure linting works and test cases are existing. The current implementation cannot be accepted, there are too many wrong source code lines, I will provide a detailed review soon.

File: hicexplorer/hicAggregateContacts.py [hunk 1]
```diff
@@ -1,7 +1,5 @@
 from __future__ import division
-import warnings
-warnings.simplefilter(action="ignore", category=RuntimeWarning)
-warnings.simplefilter(action="ignore", category=PendingDeprecationWarning)
+
```
Comment (joachimwolff) [inline hicexplorer/hicAggregateContacts.py]: Don't delete it!

File: hicexplorer/hicAggregateContacts.py [hunk 2]
```diff
@@ -12,11 +10,10 @@
 import matplotlib.pyplot as plt
 import matplotlib.gridspec as gridspec
 import matplotlib.cm as cm
-from hicmatrix import HiCMatrix as hm
+import hicexplorer.HiCMatrix as hm
```
Comment (joachimwolff) [inline hicexplorer/hicAggregateContacts.py]: You need to use hicmatrix, hicexplorer.HiCMatrix is no longer existing.

File: hicexplorer/hicAggregateContacts.py [hunk 3]
```diff
@@ -12,11 +10,10 @@
...
 import hicexplorer.utilities
 from .utilities import toString
 from .utilities import check_chrom_str_bytes
-from hicexplorer._version import __version__
```
Comment (joachimwolff) [inline hicexplorer/hicAggregateContacts.py]: __version__ needs to be there.

File: hicexplorer/hicAggregateContacts.py [hunk 4]
```diff
@@ -75,13 +72,23 @@ def parse_arguments(args=None):
...
+    parserOpt.add_argument('--regionReferencePosotion',
+                           help='wether centering the window around the left, the center or the right of the regions.',
+                           choices=['left', 'center', 'right'],
+                           default='left')
+
 
     parserOpt.add_argument("--help", "-h", action="help", help="show this help message and exit")
-    parserOpt.add_argument('--version', action='version',
-                           version='%(prog)s {}'.format(__version__))
+
```
Comment (joachimwolff) [inline hicexplorer/hicAggregateContacts.py]: Don't delete this parameter.

File: hicexplorer/hicAggregateContacts.py [hunk 5]
```diff
@@ -165,14 +172,11 @@ def parse_arguments(args=None):
     parserPlot.add_argument('--disable_bbox_tight',
                             help=argparse.SUPPRESS,
                             action='store_true')
-    parserOpt.add_argument('--dpi',
-                           help='Optional parameter: Resolution for the image in case the'
-                           'ouput is a raster graphics image (e.g png, jpg)',
-                           type=int,
-                           default=300)
     return parser
```
Comment (joachimwolff) [inline hicexplorer/hicAggregateContacts.py]: Don't delete dpi setting.

File: hicexplorer/hicAggregateContacts.py [hunk 6]
```diff
@@ -302,25 +306,94 @@ def cluster_matrices(submatrices_dict, k, method='kmeans', how='full'):
...
+                                         0, len(order)])
+        
+        summary_plot_ax.legend(ncol=1, frameon=False, markerscale=0.5)
+    
+    cbar_x = plt.subplot(gs0[1, -1])
+    fig.colorbar(heat_fig, cax=cbar_x, orientation='vertical')
+    
+    file_name = args.diagnosticHeatmapFile.name
+    log.info('Heatmap file saved under: {}'.format(file_name))
+    plt.savefig(file_name, dpi=200, bbox_inches='tight')
```
Comment (joachimwolff) [inline hicexplorer/hicAggregateContacts.py]: Please don't use a fixed dpi level, you have args.dpi for this.

File: hicexplorer/hicAggregateContacts.py [hunk 7]
```diff
@@ -392,133 +466,67 @@ def plot_aggregated_contacts(chrom_matrix, chrom_contact_position, cluster_ids,
...
+                        else:
+                            output_matrix_name = "{file}_{chrom}_cluster_{id}.tab".format(file=args.outFilePrefixMatrix, chrom=chrom, id=cluster_number + 1)
+                        np.savetxt(output_matrix_name, chrom_avg[chrom][cluster_number], '%0.5f', delimiter='\t')
+        
         cbar_x = plt.subplot(gs[-1, idx])
         fig.colorbar(img, cax=cbar_x, orientation='horizontal')
-
+    
     if args.disable_bbox_tight:
-        plt.savefig(args.outFileName.name, dpi=args.dpi)
```
Comment (joachimwolff) [inline hicexplorer/hicAggregateContacts.py]: Again dpi

File: hicexplorer/hicAggregateContacts.py [hunk 8]
```diff
@@ -392,133 +466,67 @@ def plot_aggregated_contacts(chrom_matrix, chrom_contact_position, cluster_ids,
...
+        plt.savefig(args.outFileName.name, dpi=100)
     else:
-        plt.savefig(args.outFileName.name, dpi=args.dpi, bbox_inches='tight')
```
Comment (joachimwolff) [inline hicexplorer/hicAggregateContacts.py]: again dpi

File: hicexplorer/hicAggregateContacts.py [hunk 9]
```diff
@@ -392,133 +466,67 @@ def plot_aggregated_contacts(chrom_matrix, chrom_contact_position, cluster_ids,
...
-
+        plt.savefig(args.outFileName.name, dpi=100, bbox_inches='tight')
+    
     plt.close()
 
-
-def plot_diagnostic_heatmaps(chrom_diagonals, cluster_ids, M_half, args):
```
Comment (joachimwolff) [inline hicexplorer/hicAggregateContacts.py]: Use more or less the deleted plot_diagnostic_heatmaps function, and try to integrate your changes to it.

**Comment by gtrichard** [comment]
I'm taking care of the changes.

**Comment by gtrichard** [comment]
Solved with https://github.com/deeptools/HiCExplorer/commit/642e9fa75299fb6efa2a56e1059382c1ebbfd11a

**Pull request closed without merging**